lib/fswatcher/fswatcher.go

```diff
@@ -108,14 +119,24 @@ func (watcher *FsWatcher) watchFilesystem() {
 	}
 }
 
-func (watcher *FsWatcher) newFsEvent(eventPath string) *FsEvent {
-	if isSubpath(eventPath, watcher.folderPath) {
+func (watcher *FsWatcher) newFsEvent(eventPath string) bool {
+	if len(watcher.fsEvents) >= maxFiles {
+		watcher.debugf("Tracking too many events, aggregating: %s\n", eventPath)
+	} else if _, ok := watcher.fsEvents["."]; ok {
```

Comment:
Yes, `newFsEvent` is only called from a[ for loop](https://github.com/imsodin/syncthing/blob/ceb78b530c0e34c4a8c80512d6049722f0ef4925/lib/fswatcher/fswatcher.go#L109) in `watchFilesystem`, no concurrency. And I think running this concurrently wouldn't bring any/much benefit, as most of it had to be locked.